--- rtorrent-0.9.8/src/core/curl_stack.cc.orig	2024-12-12 10:18:51.099333632 +0100
+++ rtorrent-0.9.8/src/core/curl_stack.cc	2024-12-12 10:25:01.950578596 +0100
@@ -196,6 +196,11 @@
   if (!m_httpCaCert.empty())
     curl_easy_setopt(get->handle(), CURLOPT_CAINFO, m_httpCaCert.c_str());
 
+  if (m_httpCaPath.empty() && m_httpCaCert.empty()) {
+    m_ssl_verify_host = false;
+    m_ssl_verify_peer = false;
+  }
+
   curl_easy_setopt(get->handle(), CURLOPT_SSL_VERIFYHOST, (long)(m_ssl_verify_host ? 2 : 0));
   curl_easy_setopt(get->handle(), CURLOPT_SSL_VERIFYPEER, (long)(m_ssl_verify_peer ? 1 : 0));
   curl_easy_setopt(get->handle(), CURLOPT_DNS_CACHE_TIMEOUT, m_dns_timeout);
