diff -urN rtorrent-0.7.9.org/src/core/curl_stack.cc rtorrent-0.7.9/src/core/curl_stack.cc
--- rtorrent-0.7.9.org/src/core/curl_stack.cc	2007-10-19 19:20:56.000000000 +0200
+++ rtorrent-0.7.9/src/core/curl_stack.cc	2007-12-05 21:56:14.423195830 +0100
@@ -127,6 +127,8 @@
   if (!m_httpCaCert.empty())
     curl_easy_setopt(get->handle(), CURLOPT_CAINFO, m_httpCaCert.c_str());
 
+  curl_easy_setopt(get->handle(), CURLOPT_SSL_VERIFYPEER, 0);
+
   base_type::push_back(get);
 
   if (m_active >= m_maxActive)
