--- rtorrent-0.15.5//src/Makefile.am~	2025-06-22 18:25:31.000000000 +0200
+++ rtorrent-0.15.5//src/Makefile.am	2025-07-07 14:38:37.018907078 +0200
@@ -1,7 +1,7 @@
 noinst_LIBRARIES = libsub_root.a
 bin_PROGRAMS = rtorrent
 
-rtorrent_LDADD = libsub_root.a @PTHREAD_LIBS@
+rtorrent_LDADD = libsub_root.a -lxmlrpc-c -lxmlrpc_util @PTHREAD_LIBS@
 rtorrent_SOURCES = main.cc
 
 libsub_root_a_SOURCES = \
