--- rtorrent-0.8.7/scripts/checks.m4~	2010-06-26 14:05:04.000000000 +0200
+++ rtorrent-0.8.7/scripts/checks.m4	2010-10-27 12:50:13.294309175 +0200
@@ -209,16 +209,19 @@
     [
       if test "$withval" = "no"; then
 	AC_SEARCH_LIBS(add_wch, ncursesw,
-	  AC_DEFINE(HAVE_NCURSESW, 1, defined if ncurses wide character support is available),
-	  AC_SEARCH_LIBS(wbkgdset, ncurses curses,,echo "*** The ncurses library is required!";exit 1)
+	  [AC_DEFINE(HAVE_NCURSESW, 1, defined if ncurses wide character support is available)
+	  AC_SEARCH_LIBS(nodelay, tinfow)],
+	  AC_SEARCH_LIBS(wbkgdset, ncurses curses,AC_SEARCH_LIBS(nodelay, tinfo),echo "*** The ncurses library is required!";exit 1)
         )
       else
 	AC_SEARCH_LIBS(wbkgdset, ncurses curses,,echo "*** The ncurses library is required!";exit 1)
+        AC_SEARCH_LIBS(nodelay, tinfo)
       fi
     ],[
       AC_SEARCH_LIBS(add_wch, ncursesw,
-	AC_DEFINE(HAVE_NCURSESW, 1, defined if ncurses wide character support is available),
-	AC_SEARCH_LIBS(wbkgdset, ncurses curses,,echo "*** The ncurses library is required!";exit 1)
+	[AC_DEFINE(HAVE_NCURSESW, 1, defined if ncurses wide character support is available)
+	AC_SEARCH_LIBS(nodelay, tinfow)],
+	AC_SEARCH_LIBS(wbkgdset, ncurses curses,AC_SEARCH_LIBS(nodelay, tinfow),echo "*** The ncurses library is required!";exit 1)
       )
     ])
 ])
--- rtorrent-0.8.9/scripts/checks.m4~	2011-09-02 19:46:02.000000000 +0200
+++ rtorrent-0.8.9/scripts/checks.m4	2011-09-02 20:10:27.597473502 +0200
@@ -448,8 +448,8 @@
       fi
       
       if eval $xmlrpc_cc_prg --version 2>/dev/null >/dev/null; then
-        CXXFLAGS="$CXXFLAGS `$xmlrpc_cc_prg --cflags server-util`"
-        LIBS="$LIBS `$xmlrpc_cc_prg server-util --libs`"
+        CXXFLAGS="$CXXFLAGS `$xmlrpc_cc_prg --cflags server-util xmlrpc_util`"
+        LIBS="$LIBS `$xmlrpc_cc_prg server-util --libs` -lxmlrpc-c -lxmlrpc_util"
 
         AC_TRY_LINK(
         [ #include <xmlrpc-c/server.h>
